src/logger.js

```diff
@@ -0,0 +1,39 @@
+const chalk = require('chalk');
+
+const messageDetails = (details) => {
+  if (typeof details !== 'object') {
+    return { message: details };
+  }
+
+  return { context: details.context, message: details.message || '' };
+};
+
+const getMessage = (details, color) => {
+  const { context, message } = messageDetails(details);
+  let logMessage = '';
+
+  if (context) {
+    logMessage += `${chalk.bold.dim(context)} `;
+  }
+  logMessage += chalk[color](message);
+
+  return logMessage;
+};
+
+const consoleError = details => (
+  new Error(getMessage(details, 'red'))
+);
+
+const consoleLog = details => (
+  /* eslint-disable-next-line no-console */
+  console.log(getMessage(details, 'grey'))
+);
+
+const consoleWarn = details => (
+  /* eslint-disable-next-line no-console */
+  console.warn(getMessage(details, 'magenta'))
+);
+
+module.exports = {
+  consoleError, consoleLog, consoleWarn, chalk,
```

Comment:
What are your thoughts on converting this logger to a static class?

#### Class
```
class Logger {
  static warn() {

  }

  static error() {

  }
}
```

#### Usage
```
const Logger = require('logger');
.
.
.
.
Logger.warn('This is a warning message');
.
.
.
Logger.log('This is a log message');

```

Reply:
It may also be beneficial to abstract chalk in some way. Importing and exporting a dependency directly feels a little strange. If we want to keep chalk in this file we may be able to wrap it.

#### Class
```
const chalk = require('chalk');

class Logger {
  // Code
}

Logger.chalk = chalk

module.exports = Logger;
```

#### Usage
```
const Logger = require('logger');
.
.
.
.
Logger.warn(Logger.chalk.cyan('This is a warning message'));
```

Reply:
This is overkill but if you want to be really fancy you can proxy the chalk methods onto the Logger.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy

```
const chalk = require('chalk');

class Logger {
  static log(message) {
    console.log(message);
  }
}

const LoggerProxy = {
  get: (object, property) => {
    // May want to reverse the order of lookup so Logger methods take precedence
    if (chalk[property]) {
      return chalk[property];
    }
    return Logger[property];
  },
};

module.exports = new Proxy(Logger, LoggerProxy);
```

This applies all the chalk methods onto the logger.

```
const Logger = require('./logger');
.
.
.
Logger.cyan(theme);
```

It's a little dangerous to have an unknown/pass through API, we could whitelist methods to be safer, but I thought this was really interesting either way.

Reply:
I like all of these suggestions. will look into them!